core/src/main/scala/kafka/raft/KafkaMetadataLog.scala

```diff
@@ -359,6 +360,27 @@ final class KafkaMetadataLog private (
     deleted
   }
 
+  def deleteBeforeSnapshot(snapshotId: OffsetAndEpoch,reason: SnapshotDeletionReason): Boolean = {
+    val (deleted, forgottenSnapshots) = snapshots synchronized {
+      latestSnapshotId().asScala match {
+        case Some(latestSnapshotId) if
+          snapshots.contains(snapshotId) &&
+            startOffset < snapshotId.offset &&
+            snapshotId.offset <= latestSnapshotId.offset &&
+            log.maybeIncrementLogStartOffset(snapshotId.offset, LogStartOffsetIncrementReason.SnapshotGenerated) =>
+          // Delete all segments that have a "last offset" less than the log start offset
+          val deletedSegments = log.deleteOldSegments()
+          // Remove older snapshots from the snapshots cache
+          val forgottenSnapshots = forgetSnapshotsBefore(snapshotId)
+          (deletedSegments != 0 || forgottenSnapshots.nonEmpty, forgottenSnapshots)
+        case _ =>
+          (false, mutable.TreeMap.empty[OffsetAndEpoch, Option[FileRawSnapshotReader]])
+      }
+    }
+    removeSnapshots(forgottenSnapshots, reason)
+    deleted
+  }
```

Comment:
This looks like a duplicate of `deleteBeforeSnapshot(OffsetAndEpoch)`. If tests are calling this method directly, lets add an "unknown" reason and the test can use that when deleting snapshots.

You should be able to avoid this code duplication with this change:
```scala
override def deleteBeforeSnapshot(snapshotId: OffsetAndEpoch): Boolean = {
  deleteBeforeSnapshot(snapshotId, UnknownReason())
}
```